scripts/mainmenu.js

```diff
@@ -82,37 +84,113 @@ var MainMenuController = (function () {
+					MainMenuController.onNewgameSelected(MenuContents.maps)
+					$("#MainMenuNewGameChapterDescriptionLabel").text = MenuContents.description;
+					$("#MainMenuNewGameChapterTitleLabel").text = MenuContents.name;
+					$("#MainMenuNewGameChapterTitle2Label").text = "By " + MenuContents.publisher;
+					$("#MainMenuNewGameCampaignImage").SetImage(MenuContents.icon);
+				});
+				$.CreatePanel("Label",button,"", {text: MenuContents.name, class:"CampaignButtonLabel CampaignButtonLabelPrimary"});
+				$.CreatePanel("Label",button,"", {text:"\nBy "+MenuContents.publisher, class:"CampaignButtonLabel CampaignButtonLabelSecondary"});
+				$.CreatePanel("Image",button,"", {src:MenuContents.icon, class:"CampaignButtonImage"});
+			}
+		} else {
+			//TODO: Write proper error handling for when the kv file is invalid/missing.
+			$.CreatePanel("Label", content, "", { style:"align: center center;", text: "OOPS!" });
+		}
+		_hideElement("#MainMenuButtonsNewGameContent",false)
+	}
+
+	function _onNewgameSelected(maps) {
+		let content = $("#MainMenuButtonsNewGamePanel");
+		$("#MainMenuNewGameChapterImage").SetImage("");
+		$("#MainMenuNewGamePlayButton").enabled = false;
+		content.RemoveAndDeleteChildren();
+		_hideElement("#MainNewGamePanel",false)
+		for(let [MenuIdentifier,MenuContents] of Object.entries(maps)){
+			let button = $.CreatePanel("RadioButton", content, "", {group:"RBG2", class: "ListItem MainMenuChapterListItem"});
+			button.SetPanelEvent("onactivate",(_)=>MainMenuController.onChapterSelected(MenuContents))
+			$.CreatePanel("Label",button,"", {text: MenuContents.name, class:""});
+		}
+	}
+
+	function _onChapterSelected(map) {
+		let content = $("#MainMenuNewGameChapterImage");
+		let playbutton = $("#MainMenuNewGamePlayButton");
+		playbutton.enabled = true;
+		currentSelectedMap = map.map;
+		content.SetImage(map.image);
+	}
+
+	function _playSelectedMap(){
+		if(currentSelectedMap){
+			_playMap(currentSelectedMap);
+			_hideAllSubMenus();
+			_enableMainMenu();
+		}
+	}
+
 	function _onHidePauseMenu() {
 		$("#MainMenuContainerPanel").RemoveClass("PauseMenuFade");
 		$("#BackbufferImagePanel").RemoveClass("PauseMenuVignette");
 
 		_onHomeButtonPressed();
 	}
 
+	function _enableMainMenu(bool = true) {
+		$("#MainMenuPanel").enabled = bool;
+	}
+
 	function _onHomeButtonPressed() {
 		_hideAllSubMenus();
+		_enableMainMenu();
 		$.DispatchEvent("HideContentPanel");
 	}
 
 	// --------------------------------------------------------------------------------------------------
 	// Icon buttons functions
 	// --------------------------------------------------------------------------------------------------
 
-	function _hideAllSubMenus() {
-		$("#MainMenuTopBarWorkshopContent").visible = false;
-		$("#MainMenuTopBarSettingsContent").visible = false;
-		_hideQuitMenu(true);
+	function _hideAllSubMenus() { //at this point, this function is a general "cleanup" function to get rid of any menu states/altered elements.
```

Comment:
nit: if it's a general cleanup function, can give it a better name

Reply:
It's now a general cleanup menu, we need to rewrite the way windows are handled in general, it's a mess, we shouldn't need a cleanup function, everything should clean itself. I'd say we keep it for now until we rewrite it, because currently, it just needs to work, and it isn't lying, it just gets rid of the submenus and assure the main menu is visible and clickable.